alchemist-physics/src/main/kotlin/it/unibo/alchemist/model/implementations/environments/EnvironmentWithDynamics.kt

```diff
@@ -80,12 +83,20 @@ class EnvironmentWithDynamics<T> @JvmOverloads constructor(
          * For further references: https://dyn4j.org/pages/advanced.html
          */
         world.settings.isAtRestDetectionEnabled = false
-        addGlobalReaction(PhysicsUpdate(this))
+        addGlobalReaction(physicsUpdate)
         obstacles.forEach { obstacle ->
             addObstacleToWorld(obstacle)
         }
     }
 
+    override fun addGlobalReaction(reaction: GlobalReaction<T>) {
+        if (reaction is PhysicsUpdate) {
```

Comment:
if this gets overriden more than once, throw an error. Also, add a test that checks that the error is thrown.
It's okay to redefine the physics update reaction *once*